### PR TITLE
[@types/prettier] Fix ArraySupportOption defaults

### PR DESCRIPTION
**Pull request by Shinigami92**
Followup of #47626

Context: https://github.com/prettier/plugin-pug/blob/6cb0cd36e0932316ee3ad3b0ac2dd3a94f9d4dcb/src/options/attribute-sorting/index.ts#L9

### CODE REVIEW

**Comment by Shinigami92** [comment]
@rbuckton I now used my written typedefs and found that I accidentally miss-typed this array type
Sorry for that

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## prettier ([<kbd>unpkg</kbd>](https://unpkg.com/browse/prettier@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'prettier'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. re
1. src
1. t
1. tildeTrimReplace

<details>
<summary>as well as these 180 other properties...</summary>
<p>caretTrimReplace, comparatorTrimReplace, outdent, apiDescriptor, supportsColor, commonDeprecatedHandler, commonInvalidHandler, levenUnknownHandler, createSchema, Schema, AliasSchema, AnySchema, ArraySchema, BooleanSchema, recordFromArray, mapFromArray, createAutoChecklist, partition, isInt, comparePrimitive, normalizeDefaultResult, normalizeValidateResult, normalizeDeprecatedResult, normalizeTransferResult, normalizeForwardResult, normalizeRedirectResult, ChoiceSchema, NumberSchema, IntegerSchema, StringSchema, defaultDescriptor, defaultUnknownHandler, defaultInvalidHandler, defaultDeprecatedHandler, normalize, Normalizer, matchToToken, isIdentifierStart, isIdentifierChar, isIdentifierName, isReservedWord, isStrictReservedWord, isStrictBindOnlyReservedWord, isStrictBindReservedWord, isKeyword, shouldHighlight, getChalk, codeFrameColumns, posix, win32, SEMVER_SPEC_VERSION, parse, valid, clean, SemVer, inc, diff, compareIdentifiers, rcompareIdentifiers, major, minor, patch, compare, compareLoose, rcompare, sort, rsort, gt, lt, eq, neq, gte, lte, cmp, Comparator, Range, toComparators, satisfies, maxSatisfying, minSatisfying, minVersion, validRange, ltr, gtr, outside, prerelease, intersects, coerce, minimatch, parseSync, parseString, parseFromFiles, parseFromFilesSync, flatten, splitWhen, isEnoentCodeError, createDirentFromStats, unixify, makeAbsolute, escape, removeLeadingDotSegment, isInteger, find, exceedsLimit, escapeNode, encloseBrace, isInvalidBrace, isOpenOrClose, reduce, isObject, hasRegexChars, isRegexChar, escapeRegex, toPosixSlashes, removeBackslashes, supportsLookbehinds, isWindows, escapeLast, removePrefix, wrapOutput, isStaticPattern, isDynamicPattern, convertToPositivePattern, convertToNegativePattern, isNegativePattern, isPositivePattern, getNegativePatterns, getPositivePatterns, getBaseDirectory, hasGlobStar, endsWithSlashGlobStar, isAffectDepthOfReadingPattern, expandPatternsWithBraceExpansion, expandBraceExpansion, getPatternParts, makeRe, convertPatternsToRe, matchAny, merge, isString, isEmpty, array, errno, fs, path, pattern, stream, string, generate, convertPatternsToTasks, getNegativePatternsAsPositive, groupPatternsByBaseDirectory, convertPatternGroupsToTasks, convertPatternGroupToTask, read, FILE_SYSTEM_ADAPTER, createFileSystemAdapter, Settings, stat, statSync, IS_SUPPORT_READDIR_WITH_FILE_TYPES, readdirWithFileTypes, readdir, scandir, scandirSync, isFatalError, isAppliedFilter, replacePathSegmentSeparator, joinPathSegments, walk, walkSync, walkStream, DEFAULT_FILE_SYSTEM_ADAPTER, extract, strip, parseWithComments, print, ast, code, keyword</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 03197562696a441b0fd0c4a6766d6d6b941423a3
</p>

File: types/prettier/index.d.ts [hunk 1]
```diff
@@ -365,7 +365,7 @@ export interface IntSupportOption extends BaseSupportOption<'int'> {
 }
 
 export interface IntArraySupportOption extends BaseSupportOption<'int'> {
-    default: number[];
+    default: Array<{ value: number[] }>;
```
Comment (kddeisz) [inline types/prettier/index.d.ts:368]: I'm not 100%, but I think this should be:

```ts
{ value: number }[]
```

Right? The value shouldn't itself be an array I don't think.

Comment (Shinigami92) [reply types/prettier/index.d.ts:368]: Did you see my context?
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48670#issue-500975322

**Comment by orta** [comment]
Thanks, this looks like it matches the JS

**Pull request merged**